Porting networking fixes from Java SDK; renaming IWrapper as IHasInner

martinsawicki opened the pull request:
- porting `Subnet.NetworkInterfaceIPConfigurationCount` and `.GetNetworkInterfaceIPConfigurations()` from Java SDK commit: https://github.com/Azure/azure-sdk-for-java/commit/0e61067d68badd39b7064c238fbcb08c7567c393
- porting Load Balancer child deletion workarounds for issue https://github.com/Azure/azure-sdk-for-java/issues/1456 from Java SDK commit https://github.com/Azure/azure-sdk-for-java/commit/1627af1538c9c032c38c2b0c349c6bdebdbdc8b3
- porting `NSG.NetworkInterfaceIds` return type change (Java SDK commit: https://github.com/Azure/azure-sdk-for-java/commit/b4761bcaa5ce810c82b52153d8c545e838e01516)
- renaming `IWrapper` as `IHasInner` (Java SDK commit: https://github.com/Azure/azure-sdk-for-java/commit/5e74b5627985a3c45bca98f9e649bf3aad41eb63)
- exposing `.Inner` on collections, based on Java SDK commit: https://github.com/Azure/azure-sdk-for-java/commit/5e74b5627985a3c45bca98f9e649bf3aad41eb63

Commit:
porting Subnet.NetworkInterfaceIPConfigurationCount and .GetNetworkInterfaceIPConfigurations() from Java SDK
(Java SDK commit: https://github.com/Azure/azure-sdk-for-java/commit/0e61067d68badd39b7064c238fbcb08c7567c393)

azurecla commented:
Hi __@martinsawicki__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, AZPRBOT;

Commit:
porting load balancer child deletion workarounds from Java SDK to address issue https://github.com/Azure/azure-sdk-for-java/issues/1456

Commit:
porting NSG.NetworkInterfaceIds change from Java SDK
(Java SDK Commit: https://github.com/Azure/azure-sdk-for-java/commit/b4761bcaa5ce810c82b52153d8c545e838e01516)

Commit:
renaming IWrapper as IHasInner for internal consistency and porting change from Java SDK
(part of Java SDk commit: https://github.com/Azure/azure-sdk-for-java/commit/5e74b5627985a3c45bca98f9e649bf3aad41eb63)

Commit:
file rename

Commit:
exposing Inner on collections
(ported from Java SDK commit: https://github.com/Azure/azure-sdk-for-java/commit/5e74b5627985a3c45bca98f9e649bf3aad41eb63)

hovsepm commented:
@martinsawicki merged. please reset your branch before doing next commits in thus branch.
